### PR TITLE
2038 zapier integration 1 initialize a zapier app with a twenty related account

### LINKED ISSUE #2038: Zapier integration: Initialize a zapier app with a twenty related account
Related to issue [#1043](https://github.com/orgs/twentyhq/projects/1/views/11?pane=issue&itemId=34809437)

**How**
- Create a Zapier account for Twenty
- Initialize a Zapier integration for twenty
- Commit this first app into the twenty repository (to validate with @Charles)
- Add documentation about how to update Zapier app
- Eventually add a github deployment step (to validate with @Charles)

### LINKED ISSUE #2039: Zapier integration 2: implement Zapier app authentication
Related to #2038

**How**
use the [Zapier platform cli documentation](https://platform.zapier.com/reference/cli-docs#quick-setup-guide) to create an authentication process for the Zapier App thanks to the apiKey feature

### LINKED ISSUE #2048: Zapier integration: define and implement a first trigger for Zapier app
Related to #2041

**How**
- define a first use case for a Zapier app trigger (eg: create a people when an email pops in a gmail address)
- implement that trigger

### PR DESCRIPTION
**Pull request by martmull**
- First version of the Twenty Zapier app
- has a single action for now: createPeople
- referenced privately in the Zapier directory
- testing implemented
- developer installation doc provided

### CODE REVIEW

File: docs/docs/contributor/server/basics/zapier.mdx [hunk 1]
```diff
@@ -0,0 +1,41 @@
+---
+title: Zapier App
+sidebar_position: 3
+sidebar_custom_props:
+  icon: TbBrandZapier
+---
+
+## Setup
+- Ask an admin to access the **Twenty 1Password** account
```
Comment (charlesBochet) [inline docs/docs/contributor/server/basics/zapier.mdx:9]: @martmull this should not go into the public documentation. Here we should provide a way for an external contributor to contribute: create a new Zapier integration, deploy it with the CLI using your own creds...

File: docs/docs/contributor/server/basics/zapier.mdx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+- set environment variables:
+  - In **packages/twenty-zapier** launch `cp .env.example .env`
+  - [launch backend application](/contributor/server/basics/workflows#starting-a-new-feature)
+  and go to `http://localhost:3000/graphql`
+  - set **Headers** with `{"Authorization": "Bearer YOUR_TOKEN"}`.
+  Find your token by connecting to the
+  [local front application](/contributor/frontend/basics/contributing#starting-a-new-feature)
+  and find the token in **graphql** requests' header.
+  - generate an apiKey:
+  ```
```
Comment (charlesBochet) [inline docs/docs/contributor/server/basics/zapier.mdx:30]: maybe replace this by the upcoming usage of the Settings > Developer page

File: packages/twenty-zapier/.env.example [hunk 3]
```diff
@@ -0,0 +1,2 @@
+SERVER_BASE_URL=http://localhost:3000
+ZAPIER_TEST_VALID_API_KEY=YOUR_ZAPIER_TEST_VALID_API_KEY
```
Comment (charlesBochet) [inline packages/twenty-zapier/.env.example:2]: I'm confused about the naming is this a zapier convention?
ZAPIER_API_KEY?

File: packages/twenty-zapier/authentication.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
+const test = async (z, bundle) => {
```
Comment (charlesBochet) [inline packages/twenty-zapier/authentication.js:1]: is it possible to write the whole zapier app in typescript?

File: packages/twenty-zapier/creates/create_person.js [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+        required: false,
+        list: false,
+        altersDynamicFields: false,
+      },
+    ],
+    perform: {
+      body: {
+        query:
+          `mutation 
+          CreatePerson {
```
Comment (charlesBochet) [inline packages/twenty-zapier/creates/create_person.js:56]: fix indentation?

File: packages/twenty-zapier/package.json [hunk 6]
```diff
@@ -0,0 +1,23 @@
+{
+  "name": "twenty-ui",
```
Comment (charlesBochet) [inline packages/twenty-zapier/package.json:2]: twenty-zapier!

Let's start the version at 1.0.0 for now!

**Review by charlesBochet — COMMENTED** [review]
Nice! Left a few comments! The biggest one is about using typescript instead of javascript, would that be possible?

File: packages/twenty-zapier/authentication.js (hunk 4 above)
Comment (martmull) [reply packages/twenty-zapier/authentication.js:1]: well, zapier provides [a cli](https://platform.zapier.com/quickstart/cli-tutorial) which generates `.js` files. There is some [workarounds to use typescript](https://medium.com/@Stefan1129/using-typescript-interfaces-within-your-zapier-app-c90077aede3d) but I am not sure it worths it

File: packages/twenty-zapier/creates/create_person.js (hunk 5 above)
Comment (martmull) [reply packages/twenty-zapier/creates/create_person.js:56]: yes, I need to set prettier

File: packages/twenty-zapier/package.json (hunk 6 above)
Comment (martmull) [reply packages/twenty-zapier/package.json:2]: ok

File: docs/docs/contributor/server/basics/zapier.mdx (hunk 2 above)
Comment (martmull) [reply docs/docs/contributor/server/basics/zapier.mdx:30]: ok

File: packages/twenty-zapier/authentication.js (hunk 4 above)
Comment (charlesBochet) [reply packages/twenty-zapier/authentication.js:1]: Would be great to use typescript https://github.com/zapier/zapier-platform/tree/main/example-apps/typescript

**Review by charlesBochet — APPROVED** [review]

**Pull request merged**